examples/cfd/swe_distributed_gnn/train.py

```diff
@@ -0,0 +1,580 @@
+                prd = model(inp)
+                for _ in range(cfg.data.num_future):
+                    prd = model(prd)
+                loss = l2loss(prd, tar)
+                valid_loss += loss.item() * inp.size(0)
+
+        valid_loss = valid_loss / len(dataloader.dataset)
+
+        epoch_time = time.time() - epoch_start
+
+        if metrics is not None:
+            current_lr = optimizer.param_groups[0]["lr"]
+            metrics["epochs"].append(epoch)
+            metrics["current_lr"].append(current_lr)
+            metrics["epoch_time"].append(epoch_time)
+            metrics["acc_loss"].append(acc_loss)
+            metrics["val_loss"].append(valid_loss)
+
+        scheduler.step()
+
+    train_time = time.time() - train_start
+
+    if dist_manager.rank == 0:
+        logger.info(f"Training took {train_time}.")
+        logger.info("Training Metrics")
+        for k, v in metrics.items():
+            logger.info(f"{k}: {v}")
+
+
+@hydra.main(
+    version_base="1.3", config_path=".", config_name="config_graphcast_swe.yaml"
+)
+def main(cfg: DictConfig):
+    # realistic FP32 / TF32 settings
+    torch.backends.cuda.matmul.allow_tf32 = True
+    torch.backends.cudnn.allow_tf32 = True
+    torch.set_float32_matmul_precision("high")
+
+    # default of 0.703125 gives (256, 512)
+    input_res = (
+        int(180.0 / cfg.data.angular_resolution),
+        int(360.0 / cfg.data.angular_resolution),
+    )
+
+    torch.manual_seed(cfg.seed)
+    torch.cuda.manual_seed(cfg.seed)
+
+    # initialize DistributedManager and define process group
+    # across which graph is partitioned and model is parallel across
+    # for now, based
+    DistributedManager.initialize()
+    if DistributedManager().distributed:
+        graph_partition_pg_name = "graph_partition"
+        world_size = torch.distributed.get_world_size()
+        DistributedManager.create_process_subgroup(
+            name=graph_partition_pg_name,
+            size=world_size,
+            verbose=True,
+        )
+    else:
+        world_size = 1
+        graph_partition_pg_name = None
+
+    dist_manager = DistributedManager()
+
+    hyperparameters = {
+        "world_size": world_size,
+        **OmegaConf.to_container(cfg, resolve=True),
+    }
+
+    if dist_manager.rank == 0:
+        logger.info(f"starting to run ...")
+
+    mesh_base_path = os.path.dirname(os.path.realpath(__file__))
+    meshgraph_path = os.path.join(
+        mesh_base_path, f"./icospheres_{cfg.data.mesh_size}.json"
+    )
+    model = GraphCastNet(
+        meshgraph_path=meshgraph_path,
+        static_dataset_path=None,
+        input_res=input_res,
+        input_dim_grid_nodes=3,
+        input_dim_mesh_nodes=3,
+        input_dim_edges=4,
+        output_dim_grid_nodes=3,
+        processor_layers=cfg.model.processor_layers,
+        hidden_dim=cfg.model.hidden_dim,
+        partition_size=dist_manager.group_size(graph_partition_pg_name),
```

Comment:
Yes and no. In a way, yes, group_size always should match the size of the partition group. As far as I can recall, I check for these things in the later distributed primitives.
However, it is more robust do not query the size from the process group as this would enforce the user to define the process group beforehand. In the current implementation, the user has a bit more freedom w.r.t. ordering things in his setup before the main training loop.